File: xray_scan.go
package main

import (
	"bufio"
	"os"
	"os/exec"
	"fmt"
	"syscall"
)

func main() {
	maxMemory := int64(3 * 512)

	file, err := os.Open("1.txt")
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		s := scanner.Text()
		cmd := exec.Command("./xray_linux_amd64", "ws", "--browser", fmt.Sprintf("http://%s/", s), "--html-output", fmt.Sprintf("%s.html", s))
		if err := cmd.Run(); err != nil {
			fmt.Printf("Failed to run command: %s\n", err)
		}
	}

	// Note: SetRLimit may require additional privileges
	var rlimit syscall.Rlimit
	rlimit.Max = uint64(maxMemory)
	rlimit.Cur = uint64(maxMemory)
	err = syscall.Setrlimit(syscall.RLIMIT_RSS, &rlimit)
	if err != nil {
		fmt.Printf("Error Setting Rlimit: %s", err)
		os.Exit(1)
	}
}

File: xray_scan.py
import os
import subprocess
import resource
import argparse

# Создаем парсер аргументов
parser = argparse.ArgumentParser()
parser.add_argument('-d', '--domains', help='Файл со списком доменов', required=True)
args = parser.parse_args()

max_memory = 3 * 512

# Здесь мы открываем файл, указанный в аргументе -d
with open(args.domains) as f:
    for s in map(str.strip, f):
        subprocess.run(["./xray_linux_amd64", "ws", "--browser", f"http://{s}/", "--html-output", f"{s}.html"])

resource.setrlimit(resource.RLIMIT_RSS, (max_memory, max_memory))
